src/Exercism.Representers.CSharp/Normalization/NormalizeDictionaryInitialization.cs

```diff
@@ -0,0 +1,183 @@
+                        : new KeyValuePair<SyntaxNode, SyntaxNode>(p.arg0.Syntax, p.arg1.Syntax))
+                    .ToList();
+                return (true, initializerSyntaxNodes);
+            }
+            catch (NullReferenceException nre)
+            {
+                Log.Error(nre,
+                    $"{nameof(NormalizeDictionaryInitialization)}.{nameof(ExtractInitializersWithObjectSyntax)}: dictionary initialization found with incorrect number of valid arguments (!=2)");
+                return (false, default);
+            }
+        }
+
+        private (bool Success, List<KeyValuePair<SyntaxNode, SyntaxNode>> InitializerSyntaxNodes)
+            ExtractInitializersWithCollectionSyntax(IObjectOrCollectionInitializerOperation initializerOperation)
+        {
+            string ErrorMessage(IObjectOrCollectionInitializerOperation op,
+                ImmutableArray<IArgumentOperation>? args) =>
+                args switch
+                {
+                    _ when args == default => "args is null",
+                    _ when args?.Length != 2 => $"invalid number of arguments: {args?.Length}",
+                    _ when args?[0]?.Syntax == default && args?[1]?.Syntax == default 
+                    => "args are both invalid",
+                    _ when args?[0]?.Syntax == default => "arg 0 is invalid",
+                    _ => "arg 1 is invalid"
+                } + $" in '{op.Syntax}'";
+
+            try
+            {
+                var initializerSyntaxNodes = initializerOperation.Initializers
+                    .Select(initializer => (initializer as IInvocationOperation)?.Arguments)
+                    .Select(args =>
+                        (args?.Length != 2 || args?[0]?.Syntax == default || args?[1]?.Syntax == default)
+                            ? throw new NullReferenceException(ErrorMessage(initializerOperation, args))
+                            : new KeyValuePair<SyntaxNode, SyntaxNode>(args?[0]?.Syntax, args?[1]?.Syntax)
+                    ).ToList();
+                return (true, initializerSyntaxNodes);
+            }
+            catch (NullReferenceException nre)
+            {
+                Log.Error(nre,
+                    $"{nameof(NormalizeDictionaryInitialization)}.{nameof(ExtractInitializersWithObjectSyntax)}: dictionary initialization found with incorrect number of valid arguments (!=2)");
+                return (false, default);
+            }
+        }
+
+        private string BuildDictionaryAsText(IEnumerable<KeyValuePair<SyntaxNode, SyntaxNode>> initializerArgNodes)
+        {
+            var sb = new StringBuilder();
+            sb.AppendLine("var dict = new Dictionary{");
+            foreach (var initializerArg in initializerArgNodes)
+            {
+                sb.AppendLine($"{{{initializerArg.Key.GetText()}, {initializerArg.Value.GetText()}}},");
+            }
+
+            sb.AppendLine("};");
+            return sb.ToString();
```

Comment:
It would be nice to use the Roslyn syntax factory methods to create the dictionary.

Reply:
The `dict` variable name is hardcoded in `sb.AppendLine("var dict = new Dictionary{");`

Reply:
>  It would be nice to use the Roslyn syntax factory methods to create the dictionary.

Agreed.  But I think we might be about to remove the one bit of code where a naive maintainer could say "Ah, I see whet they are doing here"!

Reply:
That's true. But if the rest is incomprehensible, it doesn't matter much anyway :D